src/bramble/api.js

```diff
@@ -0,0 +1,369 @@
+     *   }
+     *   hideUntilReady: <Boolean> whether to hide Bramble until it's fully loaded.
+     *   ready: <Function> a function to be called when Bramble is fully loaded.
+     *   provider: <Filer.FileSystem.providers.*> a provider to use for the fs, defaults to Memory
+     */
+    function Bramble(div, options) {
+        var self = this;
+
+        // Long-running callbacks for fs watch events
+        var watches = {};
+
+        // The channel port for communication with this instance
+        var port;
+
+        // The iframe's window, for postMessage
+        var brambleWindow;
+
+        // Whether to transfer ownership of ArrayBuffers or not
+        var allowArrayBufferTransfer;
+
+        if (typeof div === "object"  && !(div instanceof HTMLElement)) {
+            options = div;
+            div = null;
+        }
+        options = options || {};
+
+        var id = self._id = "bramble-" + UUID.generate();
+
+        var provider = options.provider || new Filer.FileSystem.providers.Memory();
+        var fs = self.fs = new Filer.FileSystem({provider: provider});
+
+        function startEvents(win) {
+            window.addEventListener("message", function(e) {
+                var data = parseEventData(e.data);
+
+                // When Bramble asks for initial content, reply but don't bother providing any
+                if (data.type === "bramble:init") {
+                    win.postMessage(JSON.stringify({type: "bramble:init", source: null}), "*");
+                }
+                // Listen for requests to setup the fs
+                else if (data.type === "bramble:filer") {
+                    setupChannel(win);
+                }
+                // Listen for Bramble to become ready/fully-loaded
+                else if (data.type === "bramble:loaded") {
+                    if (options.hideUntilReady) {
+                        self._iframe.style.visibility = "visible";
+                    }
+                    if (typeof options.ready === "function") {
+                        options.ready();
+                    }
+                }
+            });
+        }
+
+        function createIFrame() {
+            if (typeof div === "string") {
+                div = document.querySelector(div);
+            }
+
+            if (!div) {
+                div = document.body;
+            }
+
+            div.innerHTML = "<iframe id='" + id +
+                            "' frameborder='0' width='100%' height='100%'></iframe>";
+            
+            var iframe = self._iframe = document.getElementById(id);
+            if (options.hideUntilReady) {
+                iframe.style.visibility = "hidden";
+            }
+
+            brambleWindow = iframe.contentWindow;
+
+            startEvents(brambleWindow);
+
+            var search;
+            if (options.extensions) {
+                // Override the extension list with what's in options
```

Comment:
Yeah I guess..I only ask because your `else` condition for this copies the search string from the hosting window, so I thought it might be useful to do it in both places